src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -455,43 +457,118 @@ def create_resource(properties,
     return res.create_resource(properties, location, is_full_object)
 
 
-def show_resource(resource_group_name=None,
+def _call_on_ids(resource_ids, function, api_version, cmd_str, *args):
+    """
+    Takes care of looping through the resource_ids and calling the provided function with the right parsed arguments
+    """
+    results = []
+    for rid in resource_ids:
+        id_dict = parse_resource_id(rid)
+        if len(id_dict) == 1:
+            logger.error('az resource %s: error: argument --ids: invalid ResourceId value: \'%s\'' % (cmd_str, rid))
+            return
+        results.append(function(
+            *args,
+            resource_group_name=id_dict['resource_group'],
+            resource_provider_namespace=id_dict['resource_namespace'],
+            parent_resource_path=id_dict['resource_parent'],
+            resource_type=id_dict['resource_type'],
+            resource_name=id_dict['resource_name'], api_version=api_version))
+    return results[0] if len(results) == 1 else results
+
+
+def show_resource(resource_ids=None, resource_group_name=None,
                   resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
                   resource_name=None, api_version=None):
+    if resource_ids and isinstance(resource_ids, list):
+        return _call_on_ids(resource_ids, show_resource, api_version, "show")
```

Comment:
This just needs to be `if resource_ids`.  The way argparse works for nargs is that this will either be `None` or a list, so the second check is redundant.